x-pack/plugins/alerting/server/rules_client/rules_client.ts

```diff
@@ -631,6 +647,56 @@ export class RulesClient {
     });
   }
 
+  public async getExecutionLogForRule({
+    id,
+    dateStart,
+    dateEnd,
+    filter,
+    page,
+    perPage,
+    sort,
+  }: GetExecutionLogByIdParams): Promise<IExecutionLogResult> {
+    this.logger.debug(`getExecutionLogForRule(): getting execution log for rule ${id}`);
+    const rule = (await this.get({ id, includeLegacyId: true })) as SanitizedRuleWithLegacyId;
+
+    // Make sure user has access to this rule
+    await this.authorization.ensureAuthorized({
+      ruleTypeId: rule.alertTypeId,
+      consumer: rule.consumer,
+      operation: ReadOperations.GetExecutionLog,
+      entity: AlertingAuthorizationEntity.Rule,
+    });
+
+    // default duration of instance summary is 60 * rule interval
+    const dateNow = new Date();
+    const parsedDateStart = parseDate(dateStart, 'dateStart', dateNow);
+    const parsedDateEnd = parseDate(dateEnd, 'dateEnd', dateNow);
+
+    const eventLogClient = await this.getEventLogClient();
+
+    const results = await eventLogClient.aggregateEventsBySavedObjectIds(
+      'alert',
+      [id],
+      {
+        start: parsedDateStart.toISOString(),
+        end: parsedDateEnd.toISOString(),
+        filter,
```

Comment:
With a root filter like this it looks like we'll run into issues when filtering on a field that isn't present in all docs/sub-aggs rolling up to an aggregate execution event. This can have a couple side-effects depending on which sub-agg field the `bucket_sort`'s `sort` is against, resulting in either `1.` aggregate docs missing fields, or `2.` missing docs entirely if the `sort` is against one of those missing fields. The latter is a result of the default `gap_policy` being `skip`, and while this can be overridden to `insert_zeros`, that just puts you back in issue `1.` with missing data, and seems `gap_policy` may be removed from `bucket_sort` at some point anyway https://github.com/elastic/elasticsearch/issues/36487.

It's a bit tougher to exercise here with just the platform events since most discrete values one would filter against exist are on the main `provider:alerting AND action:execute` doc, but it showed itself pretty quick when filtering against a field that only exists on some of the custom security events.

e.g. the filter `kibana.alert.rule.execution.metrics.execution_gap_duration_s > 0` will end up filtering out all the `provider:alerting AND action:execute` docs, and so you end up with aggregate execution events missing a lot of data.


I haven't found the 'nice way' of working around this just yet ([terms filtering](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-terms-aggregation.html#_filtering_values_4) is limited and don't I see a way to circumvent the hierarchal scope when applying filters), so I am just performing a separate `terms` agg with the provided filter to first get the unique executionUUIDs, and then feeding those into this agg's query as the `filter`. If bound to `getNumExecutions` this isn't so bad, but I'm going to touch base with the ES folks to see if they have any recommendations here -- would appreciate any thoughts you may have here as well 🙂. 

Anyway, I don't think this is a particular issue this API has to deal with right now since there aren't really any fields you can sort on coming from the `alertCounts`, `actionExecution`, or `timeoutMessage` sub-aggs, but something to watch out for as additional event-logs are written with fields that aren't also included in the main alerting-execute doc.